atsign-foundation/at_client_sdk
# feat: Uptake public key hash changes

**- What I did**
- Uptake "PublicKeyHash" changes into at_client package.
   -  In "SharedKeyDecryption.dart", when decrypting the shared key, compare the hash value of the encryption public key with the hash value in atKey.metadata.pubKeyHash. If the value mismatch, throw exception.
   - Add publicKeyHash value in sync_service_impl.dart to sync the value to the remote secondary and fetch the value from the remote secondary.
   - Add  the publicKeyHash value in at_notification.metadata.

**- How to verify it**
- Updated the existing unit test to verify publicKeyHash is populated when encrypting the value.
- Updated the existing unit test to verify the exception is thrown if publicKeyHash mismatch.
- Updated the existing unit test to verify decryption is successful if publicKeyHash value matches. 

**- Description for the changelog**
- feat: Uptake public key hash changes

Note: In the pubspec.yaml, the at_persistence_secondary_server is set to "2121-uptake-public-key-hash-at-persistence_secondary-server" branch to consume the public key hash changes. Once the [PR](https://github.com/atsign-foundation/at_server/pull/2159) in at_server is merged, will replace with the hosted version.
